### PR TITLE
Boost the steemit SEO with schema markup

### PR DESCRIPTION
**Pull request by DennisAlund**
This PR is not a bug fix, it's a proposal of improvement as described in a couple of articles on steemit: 

 - [Suggestion: boost the steemit SEO with schema markup](https://steemit.com/utopian-io/@dennisalund/proposing-some-boost-to-steemit-seo-schema-markup)
 - [Looking for an SEO expert to help improving steemit schema markup](https://steemit.com/utopian-io/@dennisalund/looking-for-an-seo-expert-to-help-improving-steemit-schema-markup)

It started out from issue #2118 that I reported and PR #2119 that I submitted to fix the issues with schema.org markup. From there I figured that since the site is already using a skeleton of schema markup, it would be an improvement if it was also properly implemented with all the available attributes.

This PR is implementing schema markup for:

 - Author 
 - Comment
 - Article

And please note that steemit is hardcoded as the `publisher` for articles, since it seemed like it wasn't possible to enter the author `Person` entity as both author and publisher of the content. Seems like it needs to be an organization (described in articles)

# Current state

## Overview 
Current markup overview

## Comments
Current comment markup

## Blog
Current blog markup

# Markup proposed in this PR

## Overview
Markup overview

## Comments
Comments markup

## Blog
Blog markup

### CODE REVIEW

File: src/app/components/cards/Comment.jsx [hunk 1]
```diff
@@ -395,13 +396,30 @@ class CommentImpl extends React.Component {
...
+            '@id': comment_link,
+            datePublished: comment.created,
+            text: comment.body,
+            keywords: comment.category,
+            /*
+            // Comment's parent item must be of schema.org/Question type ... O_o
+            "parentItem": {
+                "@id": comment.parent_permlink
+            },
+            */
```
Comment (gl2748) [inline src/app/components/cards/Comment.jsx]: Agree that it is ok to ommit this.

File: src/app/components/cards/PostFull.jsx [hunk 2]
```diff
@@ -441,16 +441,49 @@ class PostFull extends React.Component {
...
+                '@id': `https://${APP_DOMAIN}${content.url}`,
+                headline: content.title,
+                text: content.body,
+                url: `https://${APP_DOMAIN}${content.url}`,
+                keywords: content.category,
+                datePublished: content.created,
+                dateModified: content.created,
+                mainEntityOfPage: `https://${APP_DOMAIN}${content.url}`,
+                image: jsonMetadata.image,
+                publisher: {
```
Comment (gl2748) [inline src/app/components/cards/PostFull.jsx]: Yes it does make sense that steemit is attributed as the publisher of the blogPosting.

Comment (DennisAlund) [reply src/app/components/cards/PostFull.jsx]: The documentation of schema.org/Publisher is unfortunately not consistent with the validation tool. According to the documentation, it should be allowed to be a `Person` but the reality looks like this

Publisher can not be a Person

There are [discussions around this](https://productforums.google.com/forum/#!topic/webmasters/hhpenBFwka4) that might suggest that it's only for AMP, which steemit doesn't have anyway. Google seem to suggest that there's far less mandatory fields for non-AMP: https://developers.google.com/search/docs/data-types/article#non-amp-sd

I'd love to get some feedback from anyone with good experience of SEO with schema.org that could confirm this claim. I'll set the `publisher` attribute to the same as `author` in the meanwhile.

**Comment by DennisAlund** [comment]
A bit of heads up regarding the result of the `author` being `publisher` as described in the above comment https://github.com/steemit/condenser/pull/2170#discussion_r157675752

image

**Comment by gl2748** [comment]
Superceded by #2181

**Comment by DennisAlund** [comment]
@gl2748  please see https://github.com/steemit/condenser/pull/2181#issuecomment-352943242

**Pull request closed without merging**